src/Executor.Common/Persistence/PreparedSpentCoinsRepository.cs

```
@@ -56,15 +56,21 @@ public async Task AddOrIgnoreAsync(IReadOnlyCollection<SpentCoin> spentCoins)
         {
             await using var context = new DatabaseContext(_dbContextOptionsBuilder.Options);
             var entities = spentCoins.Select(MapToEntity);
-
+            var transaction = await context.Database.BeginTransactionAsync();
             try
             {
                 context.PreparedSpentCoins.AddRange(entities);
 
                 await context.SaveChangesAsync();
+                await transaction.CommitAsync();
             }
             catch (DbUpdateException e) when (e.InnerException is PostgresException pgEx && pgEx.SqlState == "23505")
             {
+                await transaction.RollbackAsync();
+            }
+            catch (Exception)
+            {
+                await transaction.RollbackAsync();
```

we silently swallow all exceptions here... rethrow it